crates/core/component/stake/src/component/stake.rs

```diff
@@ -133,22 +133,20 @@ impl Component for Staking {
             .expect("should be able to track uptime");
     }
 
+    /// Writes the delegation changes for this block.
     #[instrument(name = "staking", skip(state, end_block))]
     async fn end_block<S: StateWrite + 'static>(
         state: &mut Arc<S>,
         end_block: &abci::request::EndBlock,
     ) {
         let state = Arc::get_mut(state).expect("state should be unique");
-        // Write the delegation changes for this block.
-        state
-            .set_delegation_changes(
-                end_block
-                    .height
-                    .try_into()
-                    .expect("should be able to convert i64 into block height"),
-                state.get_delegation_changes_tally().clone(),
```

Comment:
this was also being cloned needlessly, see 838887d69150847ee34ada08410f735f453a5ef8